src/validators/arguments.rs

```diff
@@ -278,28 +318,55 @@ impl Validator for ArgumentsValidator {
                         Err(err) => return Err(err),
                     };
                     if !used_kwargs.contains(either_str.as_cow()?.as_ref()) {
-                        match self.var_kwargs_validator {
-                            Some(ref validator) => match validator.validate(py, value.borrow_input(), state) {
-                                Ok(value) => {
-                                    output_kwargs.set_item(either_str.as_py_string(py, state.cache_str()), value)?;
-                                }
-                                Err(ValError::LineErrors(line_errors)) => {
-                                    for err in line_errors {
-                                        errors.push(err.with_outer_location(raw_key.clone()));
+                        match self.var_kwargs_mode {
+                            VarKwargsMode::Single => match self.var_kwargs_validator {
+                                Some(ref validator) => match validator.validate(py, value.borrow_input(), state) {
```

Comment:
I prefer the newer-style "match ergonomics" over the `ref` keyword, personally.

```suggestion
                            VarKwargsMode::Single => match &self.var_kwargs_validator {
                                Some(validator) => match validator.validate(py, value.borrow_input(), state) {
```